wall_e/extensions/ban.py

```diff
@@ -361,7 +364,14 @@ async def ban(self, ctx, user: discord.Member, *args):
         )
 
         # update database
-        await BanRecord.insert_record(ban)
+        success = await BanRecord.insert_record(ban)
+        if not success:
+            self.logger.info(f"[Ban ban()] Ban for {user} is duplicate.")
+            await self.mod_channel.send(f"Ban for {user} is duplicate")
+
+        # begin purging messages from user in the last purge_window_days
+        await self.purge_messages(ctx, user, purge_window_days)
+        await ctx.message.delete()
```

Comment:
any reason why `self.ban_list.append(ban.user_id)` isn't being moved somewhere here?

Reply:
the user is added to local ban_list, dm'd they're banned, then kicked, db updated, ....
basically want the user on the watch list asap so when they're kicked they can't just join. theres no way to call this command if the user isn't in the guild so theres no way for ban to be creating a duplicate, the check is there just to be consistent i guess.

Reply:
reason it cant run without user in guild is cuz it requires an @ which won't exist if user isn't present in the guild, plus the arg is for a member not user which also prevents this from running if user isn't in the guild.

Reply:
> basically want the user on the watch list asap so when they're kicked they can't just join.

Even if they join, won't wall-e just rekick them once the `ban` command finishes, which I assume is like 3-5 seconds runtime at most? is that a big concern?

Reply:
ban command takes long to finish cuz purge_messages go through the entire guild

Reply:
hmmm, I see.

then how about a list be made called `id_for_user_messages_to_purge`.

so the `ban` command adds the banner user's ID to that list

and then you have a running thread in the background that every few seconds checks to see if that list has any ids and if it does, that background thread is responsible for deleting all those message.

similar to how the reminder cog has a background thread polling the DB for expired reminders to send out

?

Reply:
the cogs been working as is fine, this pr is just to fix the duplication bug. ban isn't used that often to have a thread running checking for messages to purge. also

> Even if they join, won't wall-e just rekick them once the ban command finishes, which I assume is like 3-5 seconds runtime at most? is that a big concern?

if they join before its on walle watch list then it wont kick them ever because the cog is only watching on joins, if it was constantly sweeping the user list that be a huge waste.

Reply:
> this pr is just to fix the duplication bug.

Fair enough, scope creep is bad.

> ban isn't used that often to have a thread running checking for messages to purge. also

maybe but during the eventual migration to slash command, given what you have said, what I proposed has to be implemented cause slash commands have to be able to respond within a matter of seconds otherwise the interaction has to be deferred which I am not a fan of.

> if they join before its on walle watch list then it wont kick them ever because the cog is only watching on joins

Fair point, forgot about that.

Reply:
just dont convert to a slash command. i think its perfectly fine to have non slash commands especially if they're commands that mods are using. but this is a topic for another time.

Reply:
> but this is a topic for another time.

agreed and I was planning on asking mods who care about wall-e their thoughts later on.